topdown/eval.go

```diff
@@ -3044,18 +3042,20 @@ func (e *evalEvery) save(iter unifyIterator) error {
 	cpy := e.expr.Copy()
 	every := cpy.Terms.(*ast.Every)
 
-	for i := range every.Body { // TODO(sr): is there an easier way?
+	for i := range every.Body {
 		switch t := every.Body[i].Terms.(type) {
 		case *ast.Term:
-			every.Body[i].Terms = e.e.bindings.Plug(t)
+			every.Body[i].Terms = e.e.bindings.PlugNamespaced(t, e.e.caller.bindings)
 		case []*ast.Term:
-			for j := range t {
-				t[j] = e.e.bindings.Plug(t[j])
+			for j := 1; j < len(t); j++ { // don't plug operator, t[0]
+				t[j] = e.e.bindings.PlugNamespaced(t[j], e.e.caller.bindings)
 			}
 		}
 	}
```

Comment:
Unrelated to these changes, but it looks like we're only dealing with single terms and calls here. What happens if the body contains an `ast.Every`? Is there some rewriting that I'm not aware of?

Reply:
Good find. Probably an oversight. I'll look into it later.

Reply:
https://github.com/open-policy-agent/opa/issues/4826